GoWTeams.lua

```diff
@@ -436,27 +436,54 @@ function GoWTeams:AppendTeam(teamData)
                         memberContainer:AddChild(factionIcon);
 
                         local nameLabel = self.GUI:Create("Label");
-                        nameLabel:SetWidth(130);
+                        nameLabel:SetWidth(170);
                         nameLabel:SetText(member.name);
                         nameLabel:SetFontObject(GameFontNormal);
                         nameLabel:SetColor(classColorRGB.r, classColorRGB.g, classColorRGB.b);
                         memberContainer:AddChild(nameLabel);
 
+                        local roleAndIconGroup = self.GUI:Create("SimpleGroup");
+                        roleAndIconGroup:SetWidth(130);
+                        roleAndIconGroup:SetHeight(30);
+                        roleAndIconGroup:SetLayout("Flow");
+
+                        local roleIcon = self.GUI:Create("Icon");
+                        roleIcon:SetImageSize(16, 16);
+                        roleIcon:SetWidth(16);
+                        roleIcon:SetHeight(30);
+                        roleIcon:SetImage("Interface/LFGFrame/UI-LFG-ICON-PORTRAITROLES");
+                        roleIcon.image:SetPoint("TOP", roleIcon.frame, "TOP", -3, -6);
+
+                        local roleTexCoords = {
+                            [1] = { 0, 0.296875, 0.296875, 0.61 },      -- Tank
+                            [2] = { 0.296875, 0.59375, 0, 0.296875 },   -- Healer
+                            [3] = { 0.296875, 0.59375, 0.296875, 0.63 } -- DPS
+                        };
+
+                        local coords = roleTexCoords[member.specRoleId];
+                        if coords then
+                            roleIcon.image:SetTexCoord(unpack(coords));
+                        end;
```

Comment:
You can wrap this hole section inside if cords check. If icon is not found then we shouldn't add it the UI.